feat: add digest authentication (#59)
superfaceai/docs: docs/comlink/reference/provider.mdx

}
```

#### Digest Auth

This security scheme refers to HTTP Digest Authentication as described in [RFC 7616](https://datatracker.ietf.org/doc/html/rfc7616).

- `id` (string) - Unique ID for security scheme
- `type`: "http" (string)
- `scheme`: "digest" (string)

> - `statusCode` (number, optional) - HTTP status code that should be returned from initial call (containing challenge header). Default value is `401`.

Reviewer: I don't quite understand the description, let me try to rephrase it.

```suggestion
- `statusCode` (number, optional) - HTTP status code expected for the initial response with the challenge header. Default value is `401`.
```

> - `challengeHeader` (string, optional) - Case insensitive name of header containing challenge from the server. Default value is `www-authenticate`.

Reviewer: ```suggestion
- `challengeHeader` (string, optional) - Case insensitive name of the response header with server issued challenge. Default value is [`www-authenticate`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate).
```

> - `authorizationHeader` (string, optional) - Case insensitive name of header containing authorization. Default value is `Authorization`.

Reviewer: ```suggestion
- `authorizationHeader` (string, optional) - Case insensitive name of the request header containing digest authorization details. Default value is [`authorization`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization).
```
